src/backend/api/Fusion.Resources.Logic/Requests/Commands/ResourceAllocationRequest/Direct/Update.cs

```diff
@@ -0,0 +1,205 @@
+                            throw new RequestNotFoundError(request.RequestId);
+
+                        var item = await PersistChangesAsync(request, dbEntity);
+
+                        var requestItem = await mediator.Send(new GetResourceAllocationRequestItem(item.Id));
+                        return requestItem!;
+                    }
+
+                    private async Task<DbResourceAllocationRequest> PersistChangesAsync(Update request, DbResourceAllocationRequest dbItem)
+                    {
+                        bool modified = false;
+                        var updated = DateTimeOffset.UtcNow;
+
+                        if (request.AssignedDepartment.HasBeenSet)
+                        {
+                            dbItem.AssignedDepartment = request.AssignedDepartment.Value;
+                            modified = true;
+                        }
+
+                        if (request.Discipline.HasBeenSet)
+                        {
+                            dbItem.Discipline = request.Discipline.Value;
+                            modified = true;
+                        }
+
+                        if (request.ProposedPersonAzureUniqueId.HasBeenSet)
+                        {
+                            if (request.ProposedPersonAzureUniqueId.Value != null)
+                                dbItem.ProposedPerson = await profileService.EnsurePersonAsync(new PersonId(request.ProposedPersonAzureUniqueId.Value.Value));
+                            else
+                                dbItem.ProposedPerson = null;
+                            modified = true;
+                        }
+
+                        if (request.AdditionalNote.HasBeenSet)
+                        {
+                            dbItem.AdditionalNote = request.AdditionalNote.Value;
+                            modified = true;
+                        }
+
+                        if (request.ProposedChanges.HasBeenSet)
+                        {
+                            dbItem.ProposedChanges = request.ProposedChanges.Value.SerializeToString();
+                            modified = true;
+                        }
+
+                        if (request.OrgPositionInstance.HasBeenSet)
+                        {
+                            dbItem.OrgPositionInstance = request.OrgPositionInstance.Value.ToEntity();
+                            modified = true;
+
+                        }
+
+                        if (request.IsDraft.HasBeenSet)
+                        {
+                            dbItem.IsDraft = request.IsDraft.Value;
+                            modified = true;
+                        }
+
+                        /*
+                        {
+                            dbItem.ProposedPersonWasNotified =  // Should be set/reset during update when/if when notifications are enabled
+                            modified = true;
+                        }
+                        */
```

Comment:
Should ensure that there is a feature created to add notification support.

Reply:
feature created